yt/visualization/tests/test_norm_api_set_zlim_and_unit.py

```diff
@@ -0,0 +1,22 @@
+from nose.plugins.attrib import attr
+
+from yt.testing import ANSWER_TEST_TAG, fake_random_ds
+from yt.utilities.answer_testing.framework import GenericImageTest
+from yt.visualization.api import SlicePlot
+
+ds = fake_random_ds(16)
+
+
+@attr(ANSWER_TEST_TAG)
+def test_sliceplot_set_zlim_and_unit():
```

Comment:
For both this and the `test_sliceplot_set_unit_and_zlim`: could think about testing the values directly. Could do it without the need for an answer test with something like:

```python

shape = (8, 8, 8)
b = np.full(shape, 5)
b[:,:,0:5] = 10
field = ("gas", "density")
data = {field: (b, 'g/cm**3')}

ds = yt.load_uniform_grid(
    data,
    shape,
    bbox=np.array([[0, 1], [0, 1], [0, 1]]),
)

slc = yt.SlicePlot(ds, "x", field)
slc.set_zlim(field, zmin=1)
slc.set_unit(field, "kg/m**3")    
slc._setup_plots()
d = slc.plots[field]
assert(d.image.colorbar.vmin == 1000.0)

slc = yt.SlicePlot(ds, "x", field)
slc.set_unit(field, "kg/m**3")
slc.set_zlim(field, zmin=1)
slc._setup_plots()
d = slc.plots[field]
assert(d.image.colorbar.vmin == 1.0)
```

though I suppose the full image comparison test would catch issues with the colorbar label?

Reply:
Indeed the full image diff should catch this, but it doesn't hurt to have simpler assert in these tests, I think. I'll add them

Reply:
done (as separate tests)